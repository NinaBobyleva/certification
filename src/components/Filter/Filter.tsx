import { createPages } from "../../utils/createPages";
import { Button } from "../Button/Button";
import * as S from "./filter.styled";

type FilterBlockProp = {
  count: number;
  perPage: number;
  currentPage: number;
  setCurrentPage: React.Dispatch<React.SetStateAction<number>>;
};

export function Filter({
  perPage,
  currentPage,
  count,
  setCurrentPage,
}: FilterBlockProp) {
  let pages: number[] = [];

  const pagesCount = Math.ceil(count / perPage);

  createPages({ pages, pagesCount, currentPage });

  return (
    <S.FilterBox>
      {currentPage > 1 ? (
        <Button
          title="Предыдущая"
          onClick={() => setCurrentPage(currentPage - 1)}
        />
      ) : null}
      <S.FilterList>
        {pages.map((page) => (
          <S.FilterPageList
            $highlighted={currentPage === page}
            key={page}
            onClick={() => setCurrentPage(page)}
          >
            {page}
          </S.FilterPageList>
        ))}
      </S.FilterList>
      <Button
        title="Следующая"
        onClick={() => setCurrentPage(currentPage + 1)}
      />
    </S.FilterBox>
  );
}
