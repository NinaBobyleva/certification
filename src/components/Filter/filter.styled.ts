import styled from "styled-components";

interface StyledProps {
  $highlighted: boolean
}

export const FilterBox = styled.div`
  padding-top: 70px;
  padding-bottom: 40px;
  display: flex;
  align-items: center;
  gap: 25px;
`;

export const FilterList = styled.ul`
  display: flex;
  gap: 2px;
`;

export const FilterPageList = styled.li<StyledProps>`
  width: 50px;
  height: 30px;
  border: 1px solid ${props => props.$highlighted ? 'lightgrey' : '#FAFAFA'};
  border-radius: 30px;
  text-align: center;
  padding-top: 5px;
  padding-bottom: 5px;
  cursor: pointer;
  background-color: ${props => props.$highlighted ? 'lightgrey' : '#FAFAFA'};
`;
