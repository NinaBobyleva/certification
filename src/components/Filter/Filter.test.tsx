import renderer from "react-test-renderer";
import { Filter } from "./Filter";

describe("Компонент Filter", () => {
  it("Успешный рендер компонента Filter", () => {
    const setCurrentPage = jest.fn();
    const component = renderer
      .create(<Filter count={7542} perPage={10} currentPage={1} setCurrentPage={setCurrentPage} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});