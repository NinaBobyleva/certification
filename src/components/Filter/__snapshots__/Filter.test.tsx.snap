// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Button Успешный рендер Button 1`] = `
<div
  className="sc-gtLWhw bgSXdt"
>
  <ul
    className="sc-egkSDF eexUqU"
  >
    <li
      className="sc-fAUdSK kQWOkT"
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="sc-fAUdSK hmzUPb"
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="sc-fAUdSK hmzUPb"
      onClick={[Function]}
    >
      3
    </li>
    <li
      className="sc-fAUdSK hmzUPb"
      onClick={[Function]}
    >
      4
    </li>
    <li
      className="sc-fAUdSK hmzUPb"
      onClick={[Function]}
    >
      5
    </li>
    <li
      className="sc-fAUdSK hmzUPb"
      onClick={[Function]}
    >
      6
    </li>
    <li
      className="sc-fAUdSK hmzUPb"
      onClick={[Function]}
    >
      7
    </li>
    <li
      className="sc-fAUdSK hmzUPb"
      onClick={[Function]}
    >
      8
    </li>
    <li
      className="sc-fAUdSK hmzUPb"
      onClick={[Function]}
    >
      9
    </li>
    <li
      className="sc-fAUdSK hmzUPb"
      onClick={[Function]}
    >
      10
    </li>
  </ul>
  <button
    className="sc-blHHSb flYjWY"
    onClick={[Function]}
  >
    Следующая
  </button>
</div>
`;
